### ISSUE TITLE
Can't correctly stop drawing

### ISSUE THREAD

**Issue by vitonsky** [fabricjs/fabric.js#7969]
## Version
5.0.0

## Test Case
https://jsfiddle.net/3m51690d/

## Steps to reproduce
- Start drawing
- Wait till drawing will disabled after 3 second and only then release mouse button
- Wait when drawing will enabled and move mouse on canvas

## Expected Behavior

Not finished drawing interrupted by disabling drawing mode will disappear

## Actual Behavior

Not finished drawing will disappear only by start new drawing

## Notes

I have tried to set an empty array to `_points` property of brush, tried to call `_reset` method and tried to set new instance of Brush, but all this approaches is not work. It looks like bug of top canvas or something like this, i mean that not Brush control a drawing state.

**Comment by asturur** [comment]
I think stop drawing do require a mouse up indeed.
Your fiddle do demonstrate the issue clearly, but can i ask what is the actual use case?

**Comment by asturur** [comment]
We just need an imperative method `stopDrawing` that does not require the mouseup and that resets the internal status for drawing.

**Comment by vitonsky** [comment]
Actual use case is supporting gestures.
I want to implement 2 finger gesture while drawing. But when user start touch, fabric will handle firstly 1 finger touch and will be drawn a unnecessary point.

**Comment by vitonsky** [comment]
> We just need an imperative method stopDrawing that does not require the mouseup and that resets the internal status for drawing.

Exactly. I also suggest to implement common interface with life cycle for the Brush. To we can invoke start drawing, stop drawing, programmatically add and remove points.

However, at this time is no way for it, maybe you could suggest solution for my case? Thank you

**Comment by ShaMan123** [comment]
https://github.com/fabricjs/fabric.js/blob/e80ca902dfd4c25bb8fb9d6597a73f3668a54560/src/brushes/pencil_brush.class.js#L109

Until we/someone picks this up take a look:
https://jsfiddle.net/zcgs5xvt/

It's not perfect but you'll manage from here.
If I were you I would PR or subclass PencilBrush and add a flag that handles this case.

**Comment by ShaMan123** [comment]
I would argue that this issue points out a greater issue in design.
Canvas logic is complex because of many interactions that are possible and there isn't a proper state handling this. 
I think there should be simple methods `getState | setState` instead of so many flags they can collide and/or override each other's priority.

**Comment by vitonsky** [comment]
> https://github.com/fabricjs/fabric.js/blob/e80ca902dfd4c25bb8fb9d6597a73f3668a54560/src/brushes/pencil_brush.class.js#L109
> 
> Until we/someone picks this up take a look: https://jsfiddle.net/zcgs5xvt/
> 
> It's not perfect but you'll manage from here. If I were you I would PR or subclass PencilBrush and add a flag that handles this case.

Btw, i tried something like this, but got error `Uncaught TypeError: points[0] is undefined`, in this fiddle it's have too.

What's true way to initialize drawing and to stop it?

**Comment by vitonsky** [comment]
The error

![image](https://user-images.githubusercontent.com/86191922/170294569-43242054-4871-4422-a362-51c62916d177.png)

**Comment by ShaMan123** [comment]
Sounds like so many bugs that aren't related :)

**Comment by vitonsky** [comment]
We should consider this class of problems if will refactor brushes

**Comment by eliecerthoms** [comment]
@ShaMan123 Hi :) is there an ETA on refactoring of brushes, or do you mean that *whenever* it happens, it happens?

**Comment by ShaMan123** [comment]
Whenever... We are focused on Group and TS

**Comment by clarkzsd** [comment]
@vitonsky 
Hello, I encountered the same error as you. The current version I am using is v5.2.1. May I ask how you fixed it? Thanks
